### PR TITLE
Designation contract support

### LINKED ISSUE #265: Implement Designation wrapper class
Neo's native contracts was extended with a [Designation](https://github.com/neo-project/neo/pull/1950) contract. We should provide a wrapper class for this contract.

### PR DESCRIPTION
**Pull request by mialbu**
Closes #265

### CODE REVIEW

**Comment by gsmachado** [comment]
@mialbu what's missing in this one?

**Comment by mialbu** [comment]
@gsmachado yes, tests were missing and one parameter and return type of a method was not evident until I could test it with a preview4 node.
I've just pushed the changes and according tests and will mark the pr ready for review.

File: contract/src/test/java/io/neow3j/contract/DesignationContractTest.java [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+    @Test
+    public void testGetDesignateByRole() throws IOException {
+        setUpWireMockForCall("invokefunction", "designation_getByRole.json");
+
+        DesignationContract designationContract = new DesignationContract(neow);
+        List<ECPublicKey> list = designationContract.getDesignatedByRole(DesignationRole.STATE_VALIDATOR, 10);
+        assertThat(list, contains(account1.getECKeyPair().getPublicKey()));
+    }
+
```
Comment (csmuller) [inline contract/src/test/java/io/neow3j/contract/DesignationContractTest.java:64]: Add a test that makes sure an empty response is handled correctly when calling `getDesignatesByRole`. I.e. what happens if there are no designates when calling `getDesignatesByRole`. Needs to be tested with the neo-node. Is an empty array returned, or no stack item at all?

**Review by gsmachado — APPROVED** [review]
Overall, the `DesignationContract` class is fine. I talked to Claude and he will point you unit tests to still include.

File: contract/src/test/java/io/neow3j/contract/DesignationContractTest.java [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+        WireMock.configureFor(port);
+        neow = Neow3j.build(new HttpService("http://127.0.0.1:" + port));
+
+        account1 = new Account(ECKeyPair.create(Numeric.hexStringToByteArray(
+                "0f7d2f77f3229178650b958eb286258f0e6533d0b86ec389b862c440c6511a4b")));
+    }
+
+    @Test
+    public void testGetDesignateByRole() throws IOException {
+        setUpWireMockForCall("invokefunction", "designation_getByRole.json");
```
Comment (csmuller) [inline contract/src/test/java/io/neow3j/contract/DesignationContractTest.java:58]: Add the expected parameters to this `WireMock` setup, i.e., the integers that should be in the request. This tests if the request contains the expected content.

**Review by csmuller — CHANGES_REQUESTED** [review]

File: contract/src/test/java/io/neow3j/contract/DesignationContractTest.java (hunk 1 above)
Comment (mialbu) [reply contract/src/test/java/io/neow3j/contract/DesignationContractTest.java:64]: In that case, an empty array is returned. I'll add a test anyway. 👍🏼

**Review by csmuller — APPROVED** [review]

**Review by gsmachado — APPROVED** [review]
